iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -27,11 +27,9 @@ internal class AmqpConnectionHolder : IAmqpConnectionHolder, IAmqpUnitManager
         public AmqpConnectionHolder(IDeviceIdentity deviceIdentity)
         {
             _deviceIdentity = deviceIdentity;
-            _amqpIotConnector = new AmqpIotConnector(deviceIdentity.AmqpTransportSettings, deviceIdentity.IotHubConnectionString.HostName);
+            _amqpIotConnector = new AmqpIotConnector(deviceIdentity.AmqpTransportSettings, deviceIdentity.Options, deviceIdentity.IotHubConnectionString.HostName);
```

Comment:
But now we have to pipe client options through several classes.

Reply:
q - just a thought - do you think it'll be easier if we pipe the entire device identity instance to our amqp classes?

Reply:
I wondered about that, but chose not to change it that drastically.

Reply:
As a rule of thumb, I'd avoid passing references of "higher level" objects down to "lower level" objects. I think just passing the clientOptions bundle here should be fine

Reply:
I'd like to understand why we should avoid passing "higher level" objects down to "lower level" objects? Are you worried about lower level object inadvertently changing the state of the "higher level" instance and other objects getting a stale reference?